library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -0,0 +1,32 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface Client {
+  fun startStream(request: Request, streamCallback: StreamCallback): StreamEmitter
+}
+
+interface StreamCallback {
+  fun onHeaders(statusCode: Int, headers: Map<String, List<String>>)
+  fun onData(byteBuffer: ByteBuffer, endStream: Boolean)
+  fun onMetadata(metadata: Map<String, List<String>>, endStream: Boolean)
+  fun onTrailers(trailers: Map<String, List<String>>)
+  fun onError(error: Error)
+  fun onCanceled()
+  fun onCompletion()
+}
+
+interface StreamEmitter {
+  fun isActive(): Boolean
+  fun sendData(byteBuffer: ByteBuffer, endStream: Boolean): StreamEmitter
+  fun sendMetadata(metadata: Map<String, List<String>>, endStream: Boolean): StreamEmitter
+  
+  fun close(trailers: Map<String, List<String>>)
```

Comment:
Should this return a `Boolean`?

Reply:
It was a different take than what we were talking about. More explanation here:  https://github.com/lyft/envoy-mobile/pull/265#discussion_r305097771